src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/CSharpCodeFixVerifier`2.cs

```diff
@@ -20,62 +20,26 @@ public static class CSharpCodeFixVerifier<TAnalyzer, TCodeFix>
         where TAnalyzer : DiagnosticAnalyzer, new()
         where TCodeFix : CodeFixProvider, new()
     {
-        /// <inheritdoc cref="CodeFixVerifier{TAnalyzer, TCodeFix, TTest, TVerifier}.Diagnostic()"/>
-        public static DiagnosticResult Diagnostic()
-            => CSharpCodeFixVerifier<TAnalyzer, TCodeFix, XUnitVerifier>.Diagnostic();
-
-        /// <inheritdoc cref="CodeFixVerifier{TAnalyzer, TCodeFix, TTest, TVerifier}.Diagnostic(string)"/>
-        public static DiagnosticResult Diagnostic(string diagnosticId)
-            => CSharpCodeFixVerifier<TAnalyzer, TCodeFix, XUnitVerifier>.Diagnostic(diagnosticId);
-
-        /// <inheritdoc cref="CodeFixVerifier{TAnalyzer, TCodeFix, TTest, TVerifier}.Diagnostic(DiagnosticDescriptor)"/>
-        public static DiagnosticResult Diagnostic(DiagnosticDescriptor descriptor)
-            => CSharpCodeFixVerifier<TAnalyzer, TCodeFix, XUnitVerifier>.Diagnostic(descriptor);
-
-        /// <inheritdoc cref="CodeFixVerifier{TAnalyzer, TCodeFix, TTest, TVerifier}.VerifyAnalyzerAsync(string, DiagnosticResult[])"/>
-        public static async Task VerifyAnalyzerAsync(string source, params DiagnosticResult[] expected)
-            => await VerifyAnalyzerAsync(source, null, usePreviewLanguageVersion: true, expected);
-
-        /// <inheritdoc cref="CodeFixVerifier{TAnalyzer, TCodeFix, TTest, TVerifier}.VerifyAnalyzerAsync(string, DiagnosticResult[])"/>
-        public static async Task VerifyAnalyzerAsync(string source, ReferenceAssemblies? references, bool usePreviewLanguageVersion, params DiagnosticResult[] expected)
+        public static async Task VerifyAnalyzerAsync(string source, ReferenceAssemblies? references = null)
         {
-            Test test = new Test(references, usePreviewLanguageVersion, numberOfIterations: 1)
-            {
-                TestCode = source,
-            };
-
-            test.ExpectedDiagnostics.AddRange(expected);
-            await test.RunAsync(CancellationToken.None);
+            await VerifyCodeFixAsync(source, source, references);
```

Comment:
This looks a bit fishy, we used to have a way where we would only verify that an analyzer would throw some diagnostics, and one way to validate a codefix was produced, but now all analyzer calls will also validate the CodeFix produced the same exact source?

Reply:
@joperezr 

There are two cases:

1. `source` doesn't have any diagnostics, in this case both approaches are exactly the same
2. `source` has diagnostics, in this case the new approach is more strict. The difference is:
    - previously we will verify the diagnostics.
    - Now we will verify the diagnostic, and verify that *no codefix is offered*.

The use of `VerifyCodeFixAsync(source, source)` is actually a very common pattern in both dotnet/roslyn and dotnet/roslyn-analyzers.

Reply:
@joperezr See comments in https://github.com/dotnet/roslyn-analyzers/pull/4615.

Reply:
Oh I understand now, so this method is effectively not used now when we expect a diagnostic, instead we are now only using it when we expect no diagnostics to be thrown right? In such case, I would just rename from `VerifyAnalyzerAsync` to something like `VerifyNoDiagnosticsThrown`

Reply:
Technically, the method can be used to verify a diagnostic that has no codefix.

Currently, the codefix is capable of fixing any analyzer diagnostics, but in future, you might find cases where the analyzer reports a diagnostic that the codefix isn't capable of handling.

Just as an example, we could decide to report diagnostics in top-level, but not offer a fix (this is not true today definitely, just an example). If that happened, then this method asserts we don't offer a codefix, rather than offering a codefix that does "nothing".

By does "nothing", I mean we register the codefix (ie, it appears in lightbulb), but we immediately return the original document (selecting the code action from lightbulb will have no effect).

Reply:
Ok, fair enough.